pilot/pkg/security/authz/model/matcher/string.go

```diff
@@ -20,19 +20,22 @@ import (
 	envoy_matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
 )
 
-func StringMatcher(v string) *envoy_matcher.StringMatcher {
-	return StringMatcherWithPrefix(v, "")
+func StringMatcher(v string, v1beta1 bool) *envoy_matcher.StringMatcher {
+	return StringMatcherWithPrefix(v, "", v1beta1)
 }
 
 func StringMatcherRegex(regex string) *envoy_matcher.StringMatcher {
 	return &envoy_matcher.StringMatcher{MatchPattern: &envoy_matcher.StringMatcher_Regex{Regex: regex}}
 }
 
-func StringMatcherWithPrefix(v, prefix string) *envoy_matcher.StringMatcher {
+func StringMatcherWithPrefix(v, prefix string, v1beta1 bool) *envoy_matcher.StringMatcher {
```

Comment:
It's better to name the argument with the behavior that it does to the function, rather than how/where it is used. For that reason, maybe use smt descriptive like `treatWildcardAsRequire` rather than `v1beta1`.